### PR TITLE
Fix bot breaking when new users join groups/the team

### PR DESCRIPTION
**Pull request by StewPoll**
This is a resurrection of #117, which was designed to fix the problems with #115 and #61 

Test coverage should be possible. I'm  not sure exactly what tests you'd want though. This lives within the `MessageDispatcher` classes `loop` function, but it primarily relies on `SlackClient` for actual functionality, so if anything a test for `parse_channel_data` should be done.

### CODE REVIEW

File: slackbot/dispatcher.py [hunk 1]
```diff
@@ -138,9 +138,14 @@ def loop(self):
         while True:
             events = self._client.rtm_read()
             for event in events:
-                if event.get('type') != 'message':
-                    continue
-                self._on_new_message(event)
+                if event.get('type') == 'message':
```
Comment (jtatum) [inline slackbot/dispatcher.py]: rather than repeatedly calling .get(), why not store the type first, like event_type = event.get('type')

**Review by jtatum — CHANGES_REQUESTED** [review]
Thank you so much for recreating the PR, I appreciate it! Tests are possible, but I can see that they might be challenging. Let's get this merged and I'll take a crack at them. Can you briefly detail the testing you've performed on this code? Also, were you able to run the existing functional and unit tests?

File: slackbot/dispatcher.py (hunk 1 above)
Comment (StewPoll) [reply slackbot/dispatcher.py]: That is something I can do soon for you :) Not sure if it makes it perform any better in terms of CPU/Memory though. (Probably better for CPU, worse for Memory at a rough guess?)

**Comment by StewPoll** [comment]
When it comes to testing it, a similar approach is used in the [slackrtm](https://github.com/llimllib/slackrtm/blob/master/slackrtm/client.py#L45) library. This is then used by the [Limbo slack bot](https://github.com/llimllib/limbo).

File: slackbot/dispatcher.py (hunk 1 above)
Comment (jtatum) [reply slackbot/dispatcher.py]: More about style than performance. Even on a very busy slack, neither memory nor cpu is going to be a factor with this code.

Comment (StewPoll) [reply slackbot/dispatcher.py]: That's fine. Once I've got a little bit of spare time I'll make that change for you :)

**Review by jtatum — APPROVED** [review]

**Comment by tehranian** [comment]
Thanks guys for fixing this! When are you planning to ship a release to PyPI?

**Comment by jtatum** [comment]
@tehranian, I haven't heard from @lins05 in a while, but I think I can cut a release. I still want to add some tests for this code, and I need to make a changelog. Maybe next weekend? In the meantime, feel free to use setup.py install to install the version from the develop branch on github.

**Pull request merged**